warehouse/query-core/src/main/java/datawave/query/util/ssdeep/NGramScoreTuple.java

```diff
@@ -1,48 +1,62 @@
 package datawave.query.util.ssdeep;
 
 import java.io.Serializable;
-import java.util.Objects;
 
 public class NGramScoreTuple implements Serializable, Comparable<NGramScoreTuple> {
     final SSDeepHash ssDeepHash;
-    final float score;
+    final float baseScore;
+    final float weightedScore;
 
-    public NGramScoreTuple(SSDeepHash ssDeepHash, float score) {
+    public NGramScoreTuple(SSDeepHash ssDeepHash, float baseScore, float weightedScore) {
         this.ssDeepHash = ssDeepHash;
-        this.score = score;
+        this.baseScore = baseScore;
+        this.weightedScore = weightedScore;
     }
 
     public SSDeepHash getSsDeepHash() {
         return ssDeepHash;
     }
 
-    public float getScore() {
-        return score;
+    public float getBaseScore() {
+        return baseScore;
+    }
+
+    public float getWeightedScore() {
+        return weightedScore;
+    }
+
+    @Override
+    public String toString() {
+        return "ScoreTuple{" + "hash=" + ssDeepHash + ", baseScore=" + baseScore + ", weightedScore=" + weightedScore + '}';
     }
 
     @Override
     public boolean equals(Object o) {
         if (this == o)
             return true;
-        if (!(o instanceof NGramScoreTuple))
+        if (o == null || getClass() != o.getClass())
             return false;
+
         NGramScoreTuple that = (NGramScoreTuple) o;
-        return ssDeepHash == that.ssDeepHash && Float.compare(that.score, score) == 0;
-    }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(ssDeepHash, score);
+        if (Float.compare(baseScore, that.baseScore) != 0)
```

Comment:
Will epsilon be large enough here?

Reply:
Actually, the only place this class is created is in the transformer and the values passed in are integers.  Why is baseScore and weightedScore stored as floats in the first place?

Reply:
Refactoring artifact, will fix.